### PR TITLE
Sync dark mode on browser_info object

### PR DESCRIPTION
**Pull request by philippjfr**
Eventually useful when we want to dynamically toggle between dark and light themes.

### CODE REVIEW

File: panel/models/browser.ts [hunk 1]
```diff
@@ -4,12 +4,17 @@ import {Model} from "@bokehjs/model"
...
 
   override initialize(): void {
     super.initialize()
 
     if (window.matchMedia != null) {
-      this.model.dark_mode = window.matchMedia("(prefers-color-scheme: dark)").matches
+      const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+      darkModeMediaQuery.addEventListener('change', (e) => {
+	this.model.dark_mode = e.matches
+      });
```
Comment (philippjfr) [inline panel/models/browser.ts:16]:
```suggestion
      })
```

File: panel/models/browser.ts [hunk 2]
```diff
@@ -4,12 +4,17 @@ import {Model} from "@bokehjs/model"
 
 export class BrowserInfoView extends View {
   declare model: BrowserInfo
+  _dark_mode_listener:
```
Comment (philippjfr) [inline panel/models/browser.ts:7]:
```suggestion
```

File: panel/models/browser.ts [hunk 3]
```diff
@@ -9,7 +9,11 @@ export class BrowserInfoView extends View {
     super.initialize()
 
     if (window.matchMedia != null) {
-      this.model.dark_mode = window.matchMedia("(prefers-color-scheme: dark)").matches
+      const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+      darkModeMediaQuery.addEventListener('change', (e) => {
```
Comment (philippjfr) [inline panel/models/browser.ts:13]:
```suggestion
      darkModeMediaQuery.addEventListener("change", (e) => {
```

**Comment by MarcSkovMadsen** [comment]
Would this enable that if the user has not set `?theme=...` as query args then `pn.config.theme` is set to his/ hers browser preferences? Today its just set to `default`.

That would be a big improvement.

**Comment by MarcSkovMadsen** [comment]
> That's already possible in principle.

How?

**Comment by philippjfr** [comment]
> How?

```
pn.config.theme = 'default' if state.browser_info.dark_mode else 'default'
```

> And could it (please) be the default behaviour?

Maybe in future, but not in a patch release.

**Comment by philippjfr** [comment]
pre-commit.ci autofix

**Pull request merged**